route_guide/route_guide_server.cc

```diff
@@ -194,27 +298,35 @@ int RunServer(const std::string& db_path) {
   std::string server_address("0.0.0.0:50051");
   RouteGuideImpl service(db_path);
 
+  // Build a server using a ServerBuilder just like with gRPC.
   ServerBuilder builder;
-  builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
+  builder.AddListeningPort(
+      server_address,
+      grpc::InsecureServerCredentials());
 
+  // Unlike with gRPC, BuildAndStart() performs validation and returns a
+  // "named-tuple" with a status and server field to better handle errors.
   auto build = builder.BuildAndStart();
 
   if (!build.status.ok()) {
     std::cerr << "Failed to build and start server: "
               << build.status.error() << std::endl;
     return -1;
   }
-
+  // Finally create the server.
   std::unique_ptr<Server> server(std::move(build.server));
   std::cout << "Server listening on " << server_address << std::endl;
 
-  auto status = service.Serve(server.get());
-
-  if (!status.ok()) {
-    std::cerr << "Failed to serve: " << status.error() << std::endl;
-    return -1;
-  }
-
+  // Calling all rpc methods. Since Loop() returns the continuation, we
+  // need terminate our eventual and then start it explicitly with the Start().
```

Comment:
Small clarification:
```suggestion
  // Serve all of the RPC methods. Since Loop() returns an eventual, we
  // need terminate our eventual and then start it explicitly with Start().
```